subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/InstallationLocation.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import java.io.File;
+
+public class InstallationLocation {
+
+    private File location;
+
+    private String source;
+
+    public InstallationLocation(File location, String source) {
+        this.location = location;
+        this.source = source;
+    }
+
+    public File getLocation() {
+        return location;
+    }
+
+    public String getSource() {
```

Comment:
Why not a richer type?
An enum does not really work, but a String feels too crude

Reply:
Hmm...I'm having a hard time to come up with something richer then a String (except for a straight wrapper around a string) given its only used in error messages. Or do you intend this to be used in other scenarios?